client/cspaillier.go

```diff
@@ -6,46 +6,70 @@ import (
 	"math/big"
 )
 
-func (c *Client) CSPaillier(pubKeyPath string, m, label *big.Int) (bool, error) {
+type CSPaillierClient struct {
+	genericClient
+	encryptor *encryption.CSPaillier
+	label, m  *big.Int
+}
+
+func NewCSPaillierClient(endpoint string, pubKeyPath string, m, l *big.Int) (*CSPaillierClient, error) {
+	genericClient, err := NewGenericClient(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
 	encryptor, err := encryption.NewCSPaillierFromPubKeyFile(pubKeyPath)
 	if err != nil {
-		return false, err
+		return nil, err
 	}
 
-	c.handler.paillierEncryptor = encryptor
-	u, e, v, _ := c.handler.paillierEncryptor.Encrypt(m, label)
+	return &CSPaillierClient{
+		genericClient: *genericClient,
+		encryptor:     encryptor,
+		m:             m,
+		label:         l,
+	}, nil
+}
 
-	if err = c.openCSPaillier(m, u, e, v, label); err != nil {
-		return false, err
+func (c *CSPaillierClient) Run() error {
+	u, e, v, _ := c.encryptor.Encrypt(c.m, c.label)
+	if err := c.openCSPaillier(u, e, v); err != nil {
+		return err
 	}
 
-	challenge, err := c.cspaillierProveRandomData(u, e, label)
+	challenge, err := c.cspaillierProveRandomData(u, e)
```

Comment:
:smiley: :smiley: My bad. I wrote this on my own before @miha-stopar told me it's not meant to be interpreted this way, and then I forgot about it. I'm changing it for the sake of consistency. 

Actually, I just noticed that I forgot to rename all the functions hooked to `CSPaillierClient`! Again, for the sake of consistency (because the naming convention of functions hooked to other protocol clients), I'm applying these changes:
* Rename `openCSPaillier(...)` -> `open(...)`
* Rename `cspaillierProveRandomData(...)` -> `getProofRandomData(...)`
* Rename `cspaillierProveData(...)` -> `getProofData(...)`